sklearn/utils/estimator_checks.py

```diff
@@ -1997,3 +2083,79 @@ def check_decision_proba_consistency(name, estimator_orig):
         a = estimator.predict_proba(X_test)[:, 1]
         b = estimator.decision_function(X_test)
         assert_array_equal(rankdata(a), rankdata(b))
+
+
+def check_decision_scores_consistency(name, estimator_orig):
+    # Check that decision_function and score_samples methods of an outlier
+    # detector have outputs with perfect rank correlation.
+
+    centers = [(2, 2), (4, 4)]
+    X, _ = make_blobs(n_samples=100, random_state=0, n_features=4,
+                      centers=centers, cluster_std=1.0, shuffle=True)
+    X_test = np.random.randn(20, 2) + 4
+    estimator = clone(estimator_orig)
+    set_random_state(estimator)
+
+    estimator.fit(X)
+    a = estimator.score_samples(X_test)
+    b = estimator.decision_function(X_test)
+    assert_array_equal(rankdata(a), rankdata(b))
```

Comment:
Is this not the same as "decision_function is a translation of score_samples" above, but performed on held-out data?

Reply:
it's just tested here now. LGTM

Reply:
remove it. All this is tested in check_outliers_train